src/index.js

```diff
@@ -100,8 +101,47 @@ const middleware = (config = {}) => (req, res, next) => {
         error: 'Signature verification failed!'
       })
     }
+    let certificates = JSON.parse(req.headers['x-authrite-certificates'])
+    for(let cert in certificates) {
+      if(cert.subject !== identityKey) {
+        return res.status(401).json({
+          status: 'error',
+          code: 'ERR_INVALID_SUBJECT',
+          description: 'Certificate subject does not match identity key of the request sender!'
+        })
+      }
+
+      // Check valid signature
+      try {
+        authriteUtils.verifyCertificateSignature(cert)
+      } catch (err) {
+        return res.status(401).json({
+          status: 'error',
+          code: 'ERR_INVALID_CERT',
+          description: 'Invalid certificate signature'
+        })
+      }
+
+      // Check encrypted fields and decrypt them
+      let decryptedFields = {}
+      try {
+        decryptedFields = await authriteUtils.decryptCertificateFields(cert, cert.keyring, config.serverPrivateKey)
+      } catch (err) {
+        return res.status(401).json({
+          status: 'error',
+          code: 'ERR_DECRYPTION_FAILED',
+          description: 'Could not decrypt certificate fields'
+        })
+      }
+
+      return {
```

Comment:
```suggestion
      cert = {
```
We are not in a map function anymore, we can re-assign the variable to its new value, but if we `return`ed it would not proceed withthe rest of the middleware